src/bookmarklet/app.js

```diff
@@ -60,20 +58,25 @@
             // load config from inner CDN
             configOrigin = `https://${ref}--${repo}--${owner}.hlx.live`;
           }
-          const configScript = document.createElement('script');
-          configScript.id = 'hlx-sk-config';
-          configScript.src = `${configOrigin}/tools/sidekick/config.js`;
-          configScript.addEventListener('error', () => {
-            // if no project config, init sidekick with base config
-            console.info(`no sidekick config found at ${configScript.src}`);
-            window.hlx.initSidekick();
-          });
-          // init sidekick via project config
-          if (document.getElementById(configScript.id)) {
-            document.getElementById(configScript.id).replaceWith(configScript);
-          } else {
-            document.head.append(configScript);
+          try {
+            const res = await fetch(`${configOrigin}/tools/sidekick/config.json`);
+            if (res.ok) {
+              console.log('custom sidekick config found');
+              // apply custom config
+              window.hlx.sidekickConfig = {
+                ...window.hlx.sidekickConfig,
+                ...(await res.json()),
+              };
+            }
+          } catch (e) {
+            console.log('error retrieving custom sidekick config', e);
           }
+          // apply base config
+          window.hlx.sidekickConfig = {
+            ...window.hlx.sidekickConfig,
+            ...baseConfig,
```

Comment:
so the `baseConfig` wins?

Reply:
Yes: the base config contains critical properties like `owner`, `repo` and `ref`, which don't make sense to overlay in `config.json`.